sql: clean up legacy GC job code

RaduBerinde opened the pull request:
This commit cleans up legacy GC job code that is no longer used.

Informs: #112501
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114469)
<!-- Reviewable:end -->

RaduBerinde committed:
sql: clean up legacy GC job code

This commit cleans up legacy GC job code that is no longer used.

Informs: #112501
Release note: None

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 14 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114469#-:-NjJaPC_0rqaoN0zExTg:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @fqazi and @jbowens)



<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
> Reviewed 14 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114469)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jbowens)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTRs!

bors r+
